lib/overcommit/hook/pre_commit/w3c_css_validator.rb

```diff
@@ -0,0 +1,74 @@
+module Overcommit::Hook::PreCommit
+  # Runs `w3c_validators` against any modified CSS files.
+  class W3cCssValidator < Base
+    def run
+      begin
+        require 'w3c_validators'
+      rescue LoadError
+        return :warn, 'w3c_validators not installed -- run `gem install w3c_validators`'
+      end
+
+      result_messages =
+        begin
+          applicable_files.collect do |path|
+            results = validator.validate_file(path)
+            messages = results.errors + results.warnings
+            messages.collect do |msg|
+              # Some warnings are not per-line, so use 0 as a default
+              line = Integer(msg.line || 0)
+
+              # Build message by hand to reduce noise from the validator response
+              text = "#{msg.type.to_s.upcase}; URI: #{path}; line #{line}: #{msg.message.strip}"
+              Overcommit::Hook::Message.new(msg.type, path, line, text)
+            end
+          end
+        rescue W3CValidators::ValidatorUnavailable => e
+          return :fail, e.message
+        rescue W3CValidators::ParsingError => e
+          return :fail, e.message
+        end
+
+      result_messages.flatten!
+      return :pass if result_messages.empty?
```

Comment:
Hmm, well that wasn't the intent. :)

Feel free to leave as-is then. I'll circle back on this at some point. An empty array can safely be assumed to be an empty list of `Overcommit::Hook::Message`s.